### PR TITLE
SONAR-7205 WS users/search with token count in response

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: sonar-db/src/main/java/org/sonar/db/user/UserTokenDao.java [hunk 1]
```diff
@@ -53,6 +58,22 @@ public UserTokenDto selectOrFailByTokenHash(DbSession dbSession, String tokenHas
     return mapper(dbSession).selectByLogin(login);
   }
 
+  public Map<String, Integer> countTokensByLogins(final DbSession dbSession, List<String> logins) {
+    final Map<String, Integer> result = new HashMap<>();
+    DatabaseUtils.executeLargeInputs(logins, new Function<List<String>, List<UserTokenCount>>() {
+      @Override
+      public List<UserTokenCount> apply(@Nonnull List<String> input) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/user/UserTokenDao.java]: MAJOR The "input" parameter nullability is different in the superclass method, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/SearchAction.java [hunk 2]
```diff
@@ -77,31 +79,34 @@ public void handle(Request request, Response response) throws Exception {
...
+      tokenCountsByLogin = dbClient.userTokenDao().countTokensByLogins(dbSession, logins);
     } finally {
-      MyBatis.closeQuietly(dbSession);
+      dbClient.closeSession(dbSession);
     }
 
     JsonWriter json = response.newJsonWriter().beginObject();
     options.writeJson(json, result.getTotal());
-    writeUsers(json, result, fields, groupsByLogin);
+    writeUsers(json, result, groupsByLogin, tokenCountsByLogin, fields);
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/SearchAction.java]: so many parameters... but I know, it requires a deep refactoring.

**Comment by simonbrandhof** [comment]
LGTM

**Pull request closed without merging**